src/App.tsx

```diff
@@ -1,61 +1,117 @@
+import React, { useState } from 'react';
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { TodoList } from './components/TodoList';
+import { Todo } from './types/Todo';
+import { getUserId } from './services/User';
+
+const initialTodos = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUserId(todo.userId),
+}));
+
+export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>(initialTodos);
+  const [title, setTitle] = useState('');
+  const [hasTittleError, setHasTittleError] = useState(false);
+
+  const [userId, setUserId] = useState(0);
+  const [hasSelectError, setHasSelectError] = useState(false);
+
+  const addTodo = (newTodo: Omit<Todo, 'id'>) => {
+    const todoToAdd: Todo = {
+      ...newTodo,
+      id: Math.max(...todos.map(todo => todo.id)) + 1,
+    };
+
+    setTodos(currentTodos => [...currentTodos, todoToAdd]);
+  };
+
+  const handleChangeTitle = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setTitle(event.target.value);
+    setHasTittleError(false);
+  };
+
+  const handleChangeSelect = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setUserId(+event.target.value);
+    setHasSelectError(false);
+  };
+
+  const handleSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+
+    setHasTittleError(!title.trim());
+    setHasSelectError(!userId);
+
+    if (!title.trim() || !userId) {
+      return;
+    }
+
+    const newTodo = {
+      title,
+      userId,
+      completed: false,
+      user: getUserId(userId),
```

Comment:
Ensure that the `getUserId` function returns the correct user object or details as expected. If it only returns an ID, consider renaming it or adjusting its implementation to return the full user object if needed.